### PR TITLE
Ccooper/add proxy ls auth ls 12965

### PR DESCRIPTION
**Pull request by alistanis**
Adds `tcl proxy ls` and `tcl auth ls` commands. Implements #12965

Example output:

```
tctl proxy ls
./tctl proxy ls
kind: proxy
metadata:
  expires: "2022-09-07T13:31:28.531837238Z"
  id: 1662556888532240040
  name: 38c0176c-4e9b-4dcb-ac06-7409a8d74b13
spec:
  addr: 127.0.0.1:443
  hostname: cerebro
  public_addr: cerebro.alistanis.github.beta.tailscale.net:443
  rotation:
    current_id: ""
    last_rotated: "0001-01-01T00:00:00Z"
    schedule:
      standby: "0001-01-01T00:00:00Z"
      update_clients: "0001-01-01T00:00:00Z"
      update_servers: "0001-01-01T00:00:00Z"
    started: "0001-01-01T00:00:00Z"
  version: 9.3.2
version: v2
kind: proxy
metadata:
  expires: "2022-09-07T13:31:28.531837238Z"
  id: 1662556888532240040
  name: 38c0176c-4e9b-4dcb-ac06-7409a8d74b13
spec:
  addr: 127.0.0.1:443
  hostname: hephaestus
  public_addr: hephaestus.alistanis.github.beta.tailscale.net:443
  rotation:
    current_id: ""
    last_rotated: "0001-01-01T00:00:00Z"
    schedule:
      standby: "0001-01-01T00:00:00Z"
      update_clients: "0001-01-01T00:00:00Z"
      update_servers: "0001-01-01T00:00:00Z"
    started: "0001-01-01T00:00:00Z"
  version: 9.3.2
version: v2

```

```
tcl auth ls
kind: auth_server
metadata:
  expires: "2022-09-07T13:31:01.470720987Z"
  id: 1662556861470757487
  name: 38c0176c-4e9b-4dcb-ac06-7409a8d74b13
spec:
  addr: 172.31.10.175:3025
  hostname: cerebro
  rotation:
    current_id: ""
    last_rotated: "0001-01-01T00:00:00Z"
    schedule:
      standby: "0001-01-01T00:00:00Z"
      update_clients: "0001-01-01T00:00:00Z"
      update_servers: "0001-01-01T00:00:00Z"
    started: "0001-01-01T00:00:00Z"
  version: 9.3.2
version: v2
```

### CODE REVIEW

**Comment by alistanis** [comment]
Ugh please hold off on reviewing this until #14443 gets merged, I accidentally based this branch off of that one 🤦

**Comment by r0mant** [comment]
@alistanis Is this one ready for review now?

**Comment by r0mant** [comment]
@gabrielcorado @nklaassen Could you folks please review this PR when you get a chance?

File: tool/tctl/common/auth_command.go
Comment (nklaassen) [inline tool/tctl/common/auth_command.go]: you should `goimports` this, the linter is catching int

File: tool/tctl/common/proxy_command.go
Comment (nklaassen) [inline tool/tctl/common/proxy_command.go:10]: This file needs `goimports` as well

File: tool/tctl/common/proxy_command.go
Comment (nklaassen) [inline tool/tctl/common/proxy_command.go:63]: This err is not returned (linter is catching it)

File: tool/tctl/common/proxy_command.go
Comment (nklaassen) [inline tool/tctl/common/proxy_command.go:65]: See `TryRun` in `auth_command.go`, there is a default case which returns `false, nil`, and the matching cases don't return so the error is returned at the end, at the moment the `match` return value here is inverted

File: tool/tctl/common/auth_command.go
Comment (nklaassen) [inline tool/tctl/common/auth_command.go:142]: The usual default is text, any reason to go with yaml here?

File: tool/tctl/common/proxy_command_test.go
Comment (nklaassen) [inline tool/tctl/common/proxy_command_test.go:1]: is there meant to be a test here?

File: tool/tctl/common/proxy_command.go
Comment (nklaassen) [inline tool/tctl/common/proxy_command.go:1]: You'll need a license header, `make fix-license` should do the job

File: tool/tctl/common/auth_command.go
Comment (nklaassen) [inline tool/tctl/common/auth_command.go:87]: I'm assuming this is meant for testing, but I don't see any test. If we are going to introduce this I think that it should be used for every command, otherwise it's going to be confusing. Could do that in a separate PR.

File: tool/tctl/common/auth_command.go
Comment (nklaassen) [inline tool/tctl/common/auth_command.go:427]: Could you show what the output will look like in the PR description?

File: tool/tctl/common/proxy_command.go
Comment (nklaassen) [inline tool/tctl/common/proxy_command.go:47]: Could you include what the output will look like in the PR description?

File: tool/tctl/common/proxy_command.go
Comment (nklaassen) [inline tool/tctl/common/proxy_command.go:30]: the usual default is text, is there a reason to go with yaml here?

File: tool/tctl/common/proxy_command.go
Comment (gabrielcorado) [inline tool/tctl/common/proxy_command.go:44]: Should we make one flag for setting the `verbose` field? Like in the [node command](https://github.com/gravitational/teleport/blob/master/tool/tctl/common/node_command.go#L249).

File: tool/tctl/common/auth_command.go
Comment (gabrielcorado) [reply tool/tctl/common/auth_command.go:87]: If that is the case, we could consider adding it as a parameter on the [`TryRun`](https://github.com/gravitational/teleport/blob/master/tool/tctl/common/tctl.go#L85). But I agree it should be handled in a separate PR if we do this.

**Review by gabrielcorado — APPROVED** [review]
LGTM after the above comments are addressed.

**Comment by r0mant** [comment]
@alistanis Are you working on this PR?

File: tool/tctl/common/auth_command.go
Comment (alistanis) [reply tool/tctl/common/auth_command.go:87]: So for this I copied what already existed in `resource_command.go`, with the intent to finish a test for it, but I didn't get around to finishing the test before I had some family issues to attend to. In this particular case, I can remove it and look at adding another PR that covers the tests cases and adds a parameter to `TryRun`

File: tool/tctl/common/proxy_command.go
Comment (alistanis) [reply tool/tctl/common/proxy_command.go:30]: For this the GitHub issue indicated that the output should look like it already does for the other `ResourceCommand` `get` commands - which are `YAML` already: 

```
	rc.getCmd.Flag("format", "Output format: 'yaml', 'json' or 'text'").Default(teleport.YAML).StringVar(&rc.format)
```

I could change this one or both but the latter seems like a backwards incompatible change and the former would introduce an inconsistency between similar outputs. Thoughts?

**Comment by alistanis** [comment]
@nklaassen when you get a chance can you give this another look?

File: tool/tctl/common/proxy_command.go
Comment (nklaassen) [inline tool/tctl/common/proxy_command.go:84]: Semantically this should always return `true, trace.Wrap(err)` if there's a match, whether or not the error is `nil`. Most other `TryRun` implementations look like this.

```suggestion
	switch cmd {
	case p.lsCmd.FullCommand():
		err = p.ListProxies(ctx, client)
	default:
		return false, nil
	}
	return true, trace.Wrap(err)
```

**Review by nklaassen — APPROVED** [review]

**Pull request merged**
